kickstarter/android-oss
# NT-2000: Retrying posting a failed reply

# 📲 What

- Handle retrying the call to post a reply when the replying fails the first time. 
- Also handle bug that posts the reply as a root comment
- Hide the reply button on a reply comment card
- Linter cleanup

# 🛠 How

- Update the view comment replies and reply to comment streams with the new reply comment that has a parent id - we were previously sending the old comment without the updated id.

<img width="384" alt="Screen Shot 2021-07-15 at 10 16 37 AM" src="https://user-images.githubusercontent.com/19390326/125803443-a30ece81-3cab-4f78-ab65-d41eb83ff995.png">

# 👀 See


https://user-images.githubusercontent.com/19390326/125805579-6a21094f-fbe4-4496-8d56-2bdd8cfc3456.mp4


# 📋 QA

Reply on old root comment
- Open a project you have backed
- Reply to a comment on a backed project while in airplane mode
- Turn off airplane mode and retry posting the reply
- Reply should show success state "posted"
- Navigate back to root comments screen and pull to refresh a few times
- Confirm there are no duplicated comments, or that your reply was posted as a root comment
- If no replies we posted on the root comment before yours, pull to refresh should now show "view replies"
- Tap view replies and check that your reply is posted

Reply on new root comment
- Open a project you have backed
- Post a root comment in airplane mode
- Turn off airplane mode and retry posting the root comment
- Confirm success "posted" and tap reply
- Turn on airplane mode again and reply to root comment
- Turn off airplane mode and retry posting the reply
- Reply should show success state "posted"
- Navigate back to root comments screen and pull to refresh a few times
- Confirm there are no duplicated comments, or that your reply was posted as a root comment
- If no replies we posted on the root comment before yours, pull to refresh should now show "view replies"
- Tap view replies and check that your reply is posted

# Story 📖

[NT-2000: Retrying posting a failed reply](https://kickstarter.atlassian.net/browse/NT-2000)
